src/views/homepage/helpers/transform-raw-content.ts

```diff
@@ -0,0 +1,38 @@
+import { getInlineContentMaps } from 'lib/tutorials/get-inline-content-maps'
+import { formatCollectionCard } from 'components/collection-card/helpers'
+import { HomePageContentProps } from '../types'
+import { HomePageAuthoredContent } from '../content-schema'
+
+/**
+ * Given authored content for the home page,
+ * Return a prepared props object to be passed to the home page view
+ */
+async function transformRawContent(
+  authoredContent: HomePageAuthoredContent
+): Promise<HomePageContentProps> {
+  // Destructure data needed from given `authoredContent`
+  const { hero, navNotice, merchandising, learnSection, preFooter } =
+    authoredContent
+
+  // For the learnSection, transform collectionSlugs to card data
+  const { collectionSlugs, ...restLearnSection } = learnSection
+  const { inlineCollections } = await getInlineContentMaps(learnSection)
+  const collectionCards = collectionSlugs.map((slug: string) => {
+    const collectionData = inlineCollections[slug]
+    return formatCollectionCard(collectionData)
+  })
```

Comment:
Why was this part changed from what was just approved and merged in #610?

```suggestion
  // Transform collection entries into card data with `collectionSlugs` order
  const collectionCards = collectionSlugs.map((collectionSlug: string) => {
    const thisCollection = collections.find(
      (collection: Collection) => collection.slug === collectionSlug
    )
    return formatCollectionCard(thisCollection)
  })
```

Reply:
@ashleemboyer Great question, it's not a very meaningful change - functions same as it did previously. Apologies for the thrash!

Intent here was to match patterns in other pages to fill in tutorial and collection card data (such as tutorial product landing pages and the install view).

This pattern is intended to be slightly more flexible - eg if other learn content were to be added in `learnSection` or even in some other section on the home page, we'd traverse the home page content with `getInlineContentMaps` once and be able to fill in all sections (rather than having to individual reason about tutorial and collection fields, and individually fetch with `getCollections` / `getTutorials`).

Let me know if that seems reasonable, happy to adjust this if there are any strong opinions on why we should take a different approach here 🙇‍♂️ 💖

Reply:
I think until we know we need to fetch both tutorials and collections, that we should leave it as-is and not use `getInlineContentMaps` in this instance. I had opted against it in #610 because the only extra thing it does for us here is re-format the response from `getCollections` into an object from an array which doesn't feel necessary here when we're not mixing collection and tutorial cards in the same section.

Reply:
Pulling from an object instead of calling `.find()` here feels like a nice improvement. If it aligns with how we fetch learn data on other pages, the pattern will be more discoverable and consistent if we do need to adjust it down the road.

If @zchsh's approach is functionally equivalent, I don't think we need to revert here.

Reply:
I don't think there was anything wrong with the initial approach 😃, but code evolves and a new pair of eyes might see opportunities for refactoring / adjustment on another pass! 🧑‍🤝‍🧑